testing/src/main/java/io/stargate/it/grpc/ServerMetricsTest.java

```diff
@@ -0,0 +1,249 @@
+                              .contains("statusCode=\"OK\""))
+                  .anySatisfy(
+                      metric ->
+                          assertThat(metric)
+                              .contains("grpc_server_processing_duration_seconds_max")
+                              .contains("method=\"ExecuteQuery\"")
+                              .contains("methodType=\"UNARY\"")
+                              .contains("service=\"stargate.Stargate\"")
+                              .contains("statusCode=\"OK\""));
+
+              String responsesTotal =
+                  "grpc_server_responses_sent_messages_total{method=\"ExecuteQuery\",methodType=\"UNARY\",service=\"stargate.Stargate\",}";
+              String requestsTotal =
+                  "grpc_server_requests_received_messages_total{method=\"ExecuteQuery\",methodType=\"UNARY\",service=\"stargate.Stargate\",}";
+
+              assertThat(getGrpcMetric(result, responsesTotal))
+                  .hasValueSatisfying(v -> assertThat(v).isGreaterThan(0d));
+
+              assertThat(getGrpcMetric(result, requestsTotal))
+                  .hasValueSatisfying(v -> assertThat(v).isGreaterThan(0d));
+            });
+  }
+
+  @Test
+  public void batchMetrics(@TestKeyspace CqlIdentifier keyspace) {
+    StargateBlockingStub stub = stubWithCallCredentials();
+
+    Response response =
+        stub.executeBatch(
+            Batch.newBuilder()
+                .addQueries(cqlBatchQuery("INSERT INTO test (k, v) VALUES ('a', 1)"))
+                .addQueries(
+                    cqlBatchQuery(
+                        "INSERT INTO test (k, v) VALUES (?, ?)", Values.of("b"), Values.of(2)))
+                .addQueries(
+                    cqlBatchQuery(
+                        "INSERT INTO test (k, v) VALUES (?, ?)", Values.of("c"), Values.of(3)))
+                .setParameters(batchParameters(keyspace))
+                .build());
+    assertThat(response).isNotNull();
+
+    await()
+        .atMost(Duration.ofSeconds(5))
+        .untilAsserted(
+            () -> {
+              String result =
+                  RestUtils.get("", String.format("%s:8084/metrics", host), HttpStatus.SC_OK);
+
+              List<String> lines =
+                  Arrays.stream(result.split(System.getProperty("line.separator")))
+                      .filter(line -> line.startsWith("grpc_server"))
+                      .collect(Collectors.toList());
+
+              System.out.println(lines);
+              assertThat(lines)
+                  .anySatisfy(
+                      metric ->
+                          assertThat(metric)
+                              .contains("grpc_server_processing_duration_seconds_count")
+                              .contains("method=\"ExecuteBatch\"")
+                              .contains("methodType=\"UNARY\"")
+                              .contains("service=\"stargate.Stargate\"")
+                              .contains("statusCode=\"OK\""))
```

Comment:
Is it also adding HealthChecks? If not, then we should have a separate issue for adding gRPC health checks

Reply:
https://github.com/stargate/stargate/issues/1219